records/generator.go

```diff
@@ -375,37 +401,47 @@ func (rg *RecordGenerator) setFromLocal(host string, ns string) {
 	}
 }
 
-// insertRR inserts host to name's map
-// REFACTOR when storage is updated
-func (rg *RecordGenerator) insertRR(name string, host string, rtype string) {
-	logging.VeryVerbose.Println("[" + rtype + "]\t" + name + ": " + host)
-
+func (rg *RecordGenerator) exists(name string, host string, rtype string) bool {
 	if rtype == "A" {
```

Comment:
We should revisit the key datastructures holistically in order to support discoveryInfo, flexible naming, and http interface.